### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by gkanade**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by gkanade** [comment]
whitelisted sub

File: 101-hdinsight-hbase-enhancedwrite/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,150 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "clusterName": {
+      "type": "string",
+      "metadata": {
```
Comment (bmoore-msft) [inline 101-hdinsight-hbase-enhancedwrite/azuredeploy.json]: add a defaultValue with uniqueString, e.g.

"defaultValue": "[concat('hbase-', uniqueString(resourceGroup().id))]"

File: 101-hdinsight-hbase-enhancedwrite/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    },
+    "sshUserName": {
+      "type": "string",
+      "metadata": {
+        "description": "These credentials can be used to remotely access the cluster."
+      }
+    },
+    "sshPassword": {
+      "type": "securestring",
+      "metadata": {
```
Comment (bmoore-msft) [inline 101-hdinsight-hbase-enhancedwrite/azuredeploy.json]: add the minLength property of 10

File: 101-hdinsight-hbase-enhancedwrite/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      "name": "[uniqueString(resourceGroup().id)]",
+      "type": "Standard_LRS"
+    }
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[variables('defaultStorageAccount').name]",
+      "location": "[parameters('location')]",
+      "apiVersion": "2016-01-01",
```
Comment (bmoore-msft) [inline 101-hdinsight-hbase-enhancedwrite/azuredeploy.json]: update apiVersion

File: 101-hdinsight-hbase-enhancedwrite/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        "name": "[variables('defaultStorageAccount').type]"
+      },
+      "kind": "Storage",
+      "properties": {}
+    },
+    {
+      "name": "[parameters('clusterName')]",
+      "type": "Microsoft.HDInsight/clusters",
+      "location": "[parameters('location')]",
+      "apiVersion": "2015-03-01-preview",
```
Comment (bmoore-msft) [inline 101-hdinsight-hbase-enhancedwrite/azuredeploy.json]: update apiVersion

File: 101-hdinsight-hbase-enhancedwrite/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+      "dependsOn": [
+        "[concat('Microsoft.Storage/storageAccounts/',variables('defaultStorageAccount').name)]"
```
Comment (bmoore-msft) [inline 101-hdinsight-hbase-enhancedwrite/azuredeploy.json]: remove concat() and use resourceId()

File: 101-hdinsight-hbase-enhancedwrite/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+              "restAuthCredential.isEnabled": true,
+              "restAuthCredential.username": "[parameters('clusterLoginUserName')]",
+              "restAuthCredential.password": "[parameters('clusterLoginPassword')]"
+            }
+          }
+        },
+        "storageProfile": {
+          "storageaccounts": [
+            {
+              "name": "[replace(replace(reference(resourceId('Microsoft.Storage/storageAccounts', variables('defaultStorageAccount').name), '2016-01-01').primaryEndpoints.blob,'https://',''),'/','')]",
```
Comment (bmoore-msft) [inline 101-hdinsight-hbase-enhancedwrite/azuredeploy.json]: update apiVersion

File: 101-hdinsight-hbase-enhancedwrite/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+              "isDefault": true,
+              "container": "[parameters('clusterName')]",
+              "key": "[listKeys(resourceId('Microsoft.Storage/storageAccounts', variables('defaultStorageAccount').name), '2016-01-01').keys[0].value]"
```
Comment (bmoore-msft) [inline 101-hdinsight-hbase-enhancedwrite/azuredeploy.json]: update apiVersion

File: 101-hdinsight-hbase-enhancedwrite/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+    },
+    {
+      "name": "[parameters('clusterName')]",
+      "type": "Microsoft.HDInsight/clusters",
+      "location": "[parameters('location')]",
+      "apiVersion": "2015-03-01-preview",
+      "dependsOn": [
+        "[concat('Microsoft.Storage/storageAccounts/',variables('defaultStorageAccount').name)]"
+      ],
+      "tags": {},
```
Comment (bmoore-msft) [inline 101-hdinsight-hbase-enhancedwrite/azuredeploy.json]: remove empty properties

File: 101-hdinsight-hbase-enhancedwrite/azuredeploy.json (hunk 1 above)
Comment (gkanade) [reply 101-hdinsight-hbase-enhancedwrite/azuredeploy.json]: done

**Comment by gkanade** [comment]
HI @bmoore-msft I have resolved all your comments, could you please take another look ! Thx

File: 101-hdinsight-hbase-enhancedwrite/azuredeploy.json (hunk 4 above)
Comment (bmoore-msft) [reply 101-hdinsight-hbase-enhancedwrite/azuredeploy.json]: still need to update this

Comment (gkanade) [reply 101-hdinsight-hbase-enhancedwrite/azuredeploy.json]: that is the latest we have on hdi team as per our pm ashish

Comment (bmoore-msft) [reply 101-hdinsight-hbase-enhancedwrite/azuredeploy.json]: The manifest shows 2018-06-01-preview

**Pull request merged**